pkg/shared/service.go

```diff
@@ -18,6 +22,103 @@ func NewServiceHandler() *ServiceHandler {
 
 func (sh *ServiceHandler) Handle(ctx context.Context, event sdk.Event) error {
 	fmt.Println("handling object ", event.Object.GetObjectKind().GroupVersionKind().String())
+
+	sharedService := event.Object.(*v1alpha1.SharedService)
+	sharedServiceCopy := sharedService.DeepCopy()
+
+	if sharedService.Spec.ServiceType != strings.ToLower(v1alpha1.KeycloakKind) {
+		return nil
+	}
+
+	if event.Deleted {
+		return nil
+	}
+
+	if sharedService.GetDeletionTimestamp() != nil {
+		return sh.finalizeSharedService(sharedServiceCopy)
+	}
+
+	switch sharedService.Status.Phase {
+	case v1alpha1.SSPhaseNone:
+		sh.initSharedService(sharedServiceCopy)
+	case v1alpha1.SSPhaseAccepted:
+		sh.createKeycloaks(sharedServiceCopy)
+	case v1alpha1.SSPhaseComplete:
```

Comment:
I think just remove this case.

Reply:
It can if we are expecting constant monitoring of the state to be done in this PR. It's not in the ticket, so assuming there is another ticket for this.

Reply:
no, I don't expect this PR to address constant monitoring.